prometheus/metrics.go

```diff
@@ -122,15 +130,24 @@ func fetchAllMetrics(api prom_v1.API, q *IstioMetricsQuery, labels, labelsError,
 	}
 }
 
-func fetchRateRange(api prom_v1.API, metricName, labels, grouping string, q *BaseMetricsQuery) *Metric {
+func fetchRateRange(api prom_v1.API, metricName string, labels []string, grouping string, q *BaseMetricsQuery) *Metric {
 	var query string
 	// Example: round(sum(rate(my_counter{foo=bar}[5m])) by (baz), 0.001)
-	if grouping == "" {
-		query = fmt.Sprintf("sum(%s(%s%s[%s]))", q.RateFunc, metricName, labels, q.RateInterval)
-	} else {
-		query = fmt.Sprintf("sum(%s(%s%s[%s])) by (%s)", q.RateFunc, metricName, labels, q.RateInterval, grouping)
+	for i, labelsInstance := range labels {
+		if i > 0 {
+			query += " OR "
+		}
+		if grouping == "" {
+			query += fmt.Sprintf("sum(%s(%s%s[%s]))", q.RateFunc, metricName, labelsInstance, q.RateInterval)
+		} else {
+			query += fmt.Sprintf("sum(%s(%s%s[%s])) by (%s)", q.RateFunc, metricName, labelsInstance, q.RateInterval, grouping)
+		}
+	}
+	if len(labels) > 1 {
+		query = fmt.Sprintf("(%s)", query)
 	}
 	query = roundSignificant(query, 0.001)
+	fmt.Println("QUERY:\n" + query)
```

Comment:
this ^^^
Or we could as well keep this log but it should be done with the logger then

Reply:
yes, I had this here for dumping to the console while trying to update the tests.  I will remove it. 
I don't think we want to dump queries to the log either, it's too noisy, even in debug.  If you want to see the queries being generated run the prom clien tests with these Printlns.